workloadccl: two small cleanups

danhhz opened the pull request:
workloadccl: don't overwrite existing fixtures
Before this commit, this trying to store a fixture a second time
resulted in a `... already contains a BACKUP file` error. We probably
want an overwrite flag at some point, but I'll leave that for later.

workloadccl: omit runtime-only flags from fixture paths
We now don't need a new fixture when changing the TPCC mix, for example.
The metadata for which flags are runtime only is plumbed through the
workload.Flagser interface.

danhhz committed:
workloadccl: don't overwrite existing fixtures

Before this commit, this trying to store a fixture a second time
resulted in a `... already contains a BACKUP file` error. We probably
want an overwrite flag at some point, but I'll leave that for later.

Release note: None

danhhz committed:
workloadccl: omit runtime-only flags from fixture paths

We now don't need a new fixture when changing the TPCC mix, for example.
The metadata for which flags are runtime only is plumbed through the
workload.Flagser interface.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/22558)
<!-- Reviewable:end -->

danhhz commented:
@benesch Friendly ping : - )

benesch commented on pkg/ccl/workloadccl/fixture_test.go:
```diff
@@ -115,6 +115,11 @@ func TestFixture(t *testing.T) {
 
+	_, err = MakeFixture(ctx, sqlDB.DB, gcs, store, gen)
+	if !testutils.IsError(err, `already exists`) {
+		t.Fatalf(`expected 'already exists' error got: %+v`, err)
```
ooc why do you use raw string literals everywhere?

danhhz replied:
so i don't have to think about escaping when i'm reading code /shrug

danhhz committed:
workload: rename DatumSize -> ApproxDatumSize

To hint that the semantics aren't precise. Also add some commentary
while I'm here.

Release note: None

danhhz reviewed (commented):
TFTR
